das/state.go

```diff
@@ -117,20 +116,13 @@ func (s *coordinatorState) nextJob() (next job, found bool) {
 func (s *coordinatorState) nextFromPriority() (job, bool) {
 	for len(s.priority) > 0 {
 		next := s.priority[len(s.priority)-1]
+		s.priority = s.priority[:len(s.priority)-1]
 
-		// skip job if already processed
-		if next.to <= s.next {
-			s.priority = s.priority[:len(s.priority)-1]
+		// skip the job, if it will be processed next without a priority
+		if next.From == s.next {
 			continue
 		}
 
-		// cut job if it is partially processed
-		if next.from <= s.next {
```

Comment:
Why do we not need to do these checks anymore?

Reply:
Thats a good question I should have answered in changelog of previous commit. Thank you for asking it!